docs/tx-malleability.md

```diff
@@ -0,0 +1,68 @@
+
+[ADR 020](https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-020-protobuf-transaction-encoding.md#unknown-field-filtering) specifies some provisions for "non-critical" fields which can safely be ignored by older servers. In practice, I have not seen any valid usages of this. It is something in the design that maintainers should be aware of, but it may not be necessary or even 100% safe.
+
+### Non-deterministic Value Encoding (Signatures, Numbers, etc.)
+
+In addition to the non-determinism present in protobuf binary itself, some protobuf field data is encoded using a micro-format which itself may not be deterministic. Consider for instance integer or decimal encoding. Some decoders may allow for the presence of leading or trailing zeros without changing the logical meaning, ex. `00100` vs `100` or `100.00` vs `100`. So if a sign mode encodes numbers deterministically, but decoders accept multiple representations,
+a user may sign over the value `100` while `0100` gets encoded. This would be possible with Amino JSON to the extent that the integer decoder accepts leading zeros. I believe the current `Int` implementation will reject this, however, it is
+probably possible to encode a octal or hexadecimal representation in the transaction whereas the user signs over a decimal integer.
+
+Signatures themselves are encoded using a micro-format specific to the signature algorithm being used and sometimes these
+micro-formats can allow for non-determinism (multiple valid bytes for the same signature).
+I believe we have made sure that the algorithms used in the SDK reject non-canonical signature bytes.
+Any new or custom signature algorithms must make sure that they reject any non-canonical bytes, otherwise even
+with `SIGN_MODE_DIRECT` there can be transaction hash malleability by re-encoding signatures with a non-canonical
+representation.
+
+### Fields not covered by Amino JSON
+
+Another area that needs to be addressed carefully is the discrepancy between `StdSignDoc` used for `SIGN_MODE_LEGACY_AMINO_JSON` and the actual contents of `TxBody` and `AuthInfo`. If fields get added
```

Comment:
Can we link to the data structures we are talking about here?

Reply:
Would you like relative local links or github links to main or some pinned commit?

Reply:
relative local links would be great

Reply:
added in https://github.com/cosmos/cosmos-sdk/pull/23958/commits/a97fc04685fe5ce670e74a612b48130885a9a60e